src/main/java/org/apache/mesos/config/RepairConfiguration.java

```diff
@@ -0,0 +1,93 @@
+package org.apache.mesos.config;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonProperty;
+
+/**
+ * Represents the repair configuration in JSON.
+ */
+public class RepairConfiguration {
+    @Override
+    public boolean equals(Object o) {
+        if (this == o) {
+        return true;
+        }
+        if (o == null || getClass() != o.getClass()) {
+            return false;
+        }
+
+        RepairConfiguration that = (RepairConfiguration) o;
+
+        if (gracePeriodMins != that.gracePeriodMins) {
+            return false;
+        }
+        if (repairDelaySecs != that.repairDelaySecs) {
+            return false;
+        }
+        return enableReplacement == that.enableReplacement;
+
+    }
+
+    @Override
+    public int hashCode() {
+        int result = gracePeriodMins;
+        result = 31 * result + repairDelaySecs;
+        result = 31 * result + (enableReplacement ? 1 : 0);
+        return result;
+    }
+
+    @JsonProperty("repairInPlaceGracePeriodMins")
+    private int gracePeriodMins;
+    @JsonProperty("minDelayBetweenRepairsSecs")
+    private int repairDelaySecs;
+    @JsonProperty("enableReplacement")
+    private boolean enableReplacement;
+
+    public  RepairConfiguration() {}
+
+    @JsonCreator
+    public RepairConfiguration(
+            @JsonProperty("repairInPlaceGracePeriodMins") int gracePeriodMins,
+            @JsonProperty("minDelayBetweenRepairsSecs") int repairDelaySecs,
+            @JsonProperty("enableReplacement") boolean enableReplacement) {
+        this.gracePeriodMins = gracePeriodMins;
+        this.repairDelaySecs = repairDelaySecs;
+        this.enableReplacement = enableReplacement;
+    }
+
+    public boolean isEnableReplacement() {
+        return enableReplacement;
+    }
+
+    @JsonProperty("enableReplacement")
+    public void setEnableReplacement(boolean enableReplacement) {
+        this.enableReplacement = enableReplacement;
+    }
+
+    public int getRepairDelaySecs() {
```

Comment:
If serialization/deserialization is barrier, I'd suggest separating the components which are serialized from those consumed for logic.  Components which need to understand the time elements should use Duration, elements which need to serialize/deserialize should use types appropriate to that need.  Documenting in method signatures would not I think be needed for those methods which return `Duration` types.

We are already using java.time Duration here, so probalby stick with that.
https://github.com/mesosphere/dcos-commons/blob/ad7dbe51a10a768b0e05b827fa30ce42b026ec4b/src/main/java/org/apache/mesos/executor/TimedExecutorTask.java#L9